source/lac/cuda_solver_direct.cu

```diff
@@ -0,0 +1,353 @@
+                         int *pivot_dev, int *info_dev)
+    {
+      cusolverStatus_t cusolver_error_code =
+        cusolverDnSgetrf(cusolver_dn_handle, m, n, dense_matrix_dev, m,
+                         workspace_dev, pivot_dev, info_dev);
+      AssertCusolver(cusolver_error_code);
+    }
+
+
+
+    void cusolverDngetrf(cusolverDnHandle_t cusolver_dn_handle, int m, int n,
+                         double *dense_matrix_dev, double *workspace_dev,
+                         int *pivot_dev, int *info_dev)
+    {
+      cusolverStatus_t cusolver_error_code =
+        cusolverDnDgetrf(cusolver_dn_handle, m, n, dense_matrix_dev, m,
+                         workspace_dev, pivot_dev, info_dev);
+      AssertCusolver(cusolver_error_code);
+    }
+
+
+
+    void cusolverDngetrs(cusolverDnHandle_t cusolver_dn_handle, int m,
+                         float *dense_matrix_dev, int *pivot_dev, float *b,
+                         int *info_dev)
+    {
+      const int n_rhs = 1;
+      cusolverStatus_t cusolver_error_code =
+        cusolverDnSgetrs(cusolver_dn_handle, CUBLAS_OP_N, m, n_rhs,
+                         dense_matrix_dev, m, pivot_dev, b, m, info_dev);
+      AssertCusolver(cusolver_error_code);
+    }
+
+
+
+    void cusolverDngetrs(cusolverDnHandle_t cusolver_dn_handle, int m,
+                         double *dense_matrix_dev, int *pivot_dev, double *b,
+                         int *info_dev)
+    {
+      const int n_rhs = 1;
+      cusolverStatus_t cusolver_error_code =
+        cusolverDnDgetrs(cusolver_dn_handle, CUBLAS_OP_N, m, n_rhs,
+                         dense_matrix_dev, m, pivot_dev, b, m, info_dev);
+      AssertCusolver(cusolver_error_code);
+    }
+
+
+
+    void cusolverSpcsrlsvluHost(cusolverSpHandle_t cusolver_sp_handle,
+                                const unsigned int n_rows, const unsigned int nnz,
+                                cusparseMatDescr_t descr, const float *val_host,
+                                const int *row_ptr_host, const int *column_index_host,
+                                const float *b_host, float *x_host)
+    {
+      int singularity = 0;
+      cusolverStatus_t cusolver_error_code = cusolverSpScsrlsvluHost(
+                                               cusolver_sp_handle, n_rows, nnz, descr, val_host, row_ptr_host,
+                                               column_index_host, b_host, 0., 1, x_host, &singularity);
+      AssertCusolver(cusolver_error_code);
+      Assert(singularity == -1, ExcMessage("Coarse matrix is singular"));
+    }
+
+
+
+    void cusolverSpcsrlsvluHost(cusolverSpHandle_t cusolver_sp_handle,
+                                const unsigned int n_rows, unsigned int nnz,
+                                cusparseMatDescr_t descr, const double *val_host,
+                                const int *row_ptr_host,
+                                const int *column_index_host, const double *b_host,
+                                double *x_host)
```

Comment:
To ease reading, could you please break the lines in function arguments at the same place as in the `float` case?